EclEditor/LexSALT.cxx

```diff
@@ -0,0 +1,415 @@
+// Scintilla source code edit control
+/** @file LexSALT.cxx
+** Lexer for SALT.
+**/
+// Copyright 1998-2001 by Neil Hodgson <neilh@scintilla.org>
+// The License.txt file describes the conditions under which this software may be distributed.
+
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+#include <stdio.h>
+#include <stdarg.h>
+#include <assert.h>
+
+#ifdef _MSC_VER
+#pragma warning(disable: 4786)
+#endif
+#ifdef __BORLANDC__
+// Borland C++ displays warnings in vector header without this
+#pragma option -w-ccc -w-rch
+#endif
+
+#include <string>
+#include <vector>
+#include <map>
+#include <algorithm>
+
+#include "ILexer.h"
+#include "Scintilla.h"
+#include "SciLexer.h"
+
+#include "PropSetSimple.h"
+#include "WordList.h"
+#include "LexAccessor.h"
+#include "Accessor.h"
+#include "StyleContext.h"
+#include "CharacterSet.h"
+#include "LexerModule.h"
+#include "OptionSet.h"
+
+#ifdef SCI_NAMESPACE
+using namespace Scintilla;
+#endif
+
+static bool IsSpaceEquiv(int state) {
+    return (state <= SCE_SALT_COMMENT) ||
+        (state == SCE_SALT_COMMENTDOC) || (state == SCE_SALT_COMMENTLINE);
+}
+
+static std::string SaltKeywords =
+"abbr acronym add all allow "
+"also always any apposition attributefile bagofwo "
+"bagofwords basis besttype blank blocklink blockthreshold "
+"caps carry child clean cleave "
+"config csv custom "
+"dedup derived distance dive divisions docfield "
+"edit efficacy enum exclude exists "
+"explode extend externalfile fail fast fed "
+"filename first fixed flag "
+"force forces format fuzzy generate generation "
+"hack hasbase hash headersearch hybrid hyphen "
+"id idchildren iddone idfield idname "
+"idspace iduncle ignore include ingestfile initial "
+"inrange keep last lat latlong lefttrim "
+"lengths like linkpath list long many "
+"mapping match max maxspc mddm medium "
+"min mininum module most multiple named "
+"namescope never nines nofuzzy none noquotes "
+"notag nubberfield null nulls onfail only "
+"options or owned parse partition penalty "
+"permits phonetic population priority process prop "
+"quote range rangefield rds recall recorddate "
+"records recordtype reject relationship replace required "
+"ridfield rs rst scale score search "
+"segment segtype sep slow soft "
+"sourceridfield spaces split spray stem supports "
+"switch switchn tag term threshold track "
+"trigram type uber values weight weighted "
+"wheel wild within words";
```

Comment:
Actually, not doing this did not create the desired tooltip! In the NLP++ parse to generate the LangRef xml file I found 4 parameter definitions in the manual that were missing or had extra parens. I will pass this to the SALT manual team. (trivial but since I have it).